specs/WebResourceRequestedWorkers.md

```diff
@@ -0,0 +1,146 @@
+                    webResourceRequestArgs;
+                if (SUCCEEDED(args->QueryInterface(IID_PPV_ARGS(&webResourceRequestArgs))))
+                {
+                    COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE requestSource =
+                        COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_ALL;
+                    webResourceRequestArgs->get_RequestSource(&requestSource);
+                    if (requestSource != COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_ALL)
+                    {
+                        source =
+                            L", \"source\": " + WebRequestSourceToString(requestSource);
+                    }
+                }
+                std::wstring message = L"{" + WebResourceRequestedToJsonString(webResourceRequest.get(), source);
+                message += WebViewPropertiesToJsonString(m_webviewEventSource.get());
+                message += L"}";
+                PostEventMessage(message);
+                return S_OK;
+            })
+            .Get(),
+        &m_webResourceRequestedToken);
+```
+
+# API Details
+## IDL
+```c#
+/// Specifies the source of `WebResourceRequested` event.
+[v1_enum]
+typedef enum COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE {
+  /// Indicates that web resource is requested from main page including dedicated workers and iframes.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_MAIN = 1,
+
+  /// Indicates that web resource is requested from shared worker.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_SHARED_WORKER = 2,
+
+  /// Indicates that web resource is requested from service worker.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_SERVICE_WORKER = 4,
+
+  /// Indicates that web resource is requested from any supported source.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_ALL = 0XFFFFFFFF
+} COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE;
+cpp_quote("DEFINE_ENUM_FLAG_OPERATORS(COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE);")
+
+[uuid(1cc6a402-3724-4e4a-9099-8cf60f2f93a1), object, pointer_default(unique)]
+interface ICoreWebView2: IUnknown {
+  /// Extends AddWebResourceRequestedFilter to support events from shared
+  /// and service workers. To receive all raised events filters have to be added
+  /// before main page navigation.
+  /// Because service worker runs separately from main page its WebResourceRequested
+  /// will be raised for all webviews which are subscribed to
+  /// COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_SERVICE_WORKER request source.
+  /// It is recommended to enable WebResourceRequested filter for service workers
+  /// only on a main WebView to avoid handling the same WebResourceRequested
+  /// event multiple times.
+  ///
+  /// For more information about web resource requested filters, navigate to
+  /// AddWebResourceRequestedFilter.
+  HRESULT AddWebResourceRequestedFilterWithRequestSource(
+    [in] LPCWSTR const uri,
+    [in] COREWEBVIEW2_WEB_RESOURCE_CONTEXT const resourceContext,
+    [in] COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE const requestSource);
```

Comment:
What happens if requestSource is 0? We fail?

Reply:
We will not fail - just ignore it(will fix one small issue). How should we properly handle it?

Reply:
Such a filter will never match and is almost certainly an error on part of the end developer. Would be good to fail in that case (and document it)